common/to.etc.alg/src/main/java/to/etc/util/DiagnosticUtil.java

```diff
@@ -26,14 +28,17 @@
  *
  * If initLog is not called, called log methods are ignored, and report returns null result.
  */
-public class DiagnosticUtil {
+public final class DiagnosticUtil {
 
 	private static ThreadLocal<Map<String, Object>> diagnostic = new ThreadLocal<>();
 	private static ThreadLocal<Boolean> diagnosticEnabled = new ThreadLocal<>();
 
 	private static final String SUB_MAP = "_subMap";
 
-	private static final SimpleDateFormat DF = new SimpleDateFormat("HH:mm:ss.SSS");
+	@NonNull
+	private static final ThreadLocal<SimpleDateFormat> DF = new ThreadLocal<>().withInitial(() -> new SimpleDateFormat("HH:mm:ss.SSS"));
```

Comment:
We do not really do this anymore. Creating a SimpleDateFormat is a pity but as long as you do not do it in a loop for 213132241324 times it is pretty cheap. And in other code we use FastDateFormat.
These ThreadLocals are being complained about when Tomcat toggles a server..